docker/worker/worker.py

```diff
@@ -460,11 +460,38 @@ def _analyze_vulnerability(self, source_repo, repo, vulnerability, path,
       if fix_result:
         added_fix_info = add_fix_information(vulnerability, fix_result)
 
-    result = osv.analyze(
-        vulnerability,
-        analyze_git=not source_repo.ignore_git,
-        detect_cherrypicks=source_repo.detect_cherrypicks,
-        versions_from_repo=source_repo.versions_from_repo)
+    analyze_git = not source_repo.ignore_git
+
+    try:
+      result = osv.analyze(
+          vulnerability,
+          analyze_git=analyze_git,
+          detect_cherrypicks=source_repo.detect_cherrypicks,
+          versions_from_repo=source_repo.versions_from_repo)
+    except (IndexError, subprocess.CalledProcessError):
```

Comment:
some questions:

1. Can we re-raise a more specific error closer to where the actual commit issue is detected? 

2. Does it make sense to only strip the offending commit, as opposed to stripping every single GIT commit? It may also be easy to do this if we do 1 and propagate the necessary context via our own exception type.

Reply:
> Can we re-raise a more specific error closer to where the actual commit issue is detected?

Yes, is it okay to change the behaviour of `RepoAnalyzer`?

 > Does it make sense to only strip the offending commit, as opposed to stripping every single GIT commit? It may also be easy to do this if we do 1 and propagate the necessary context via our own exception type.

That would be more ideal, I'm not sure how easy it will be to identify which one it is, but I can see how we go...

Reply:
> > Can we re-raise a more specific error closer to where the actual commit issue is detected?
> 
> Yes, is it okay to change the behaviour of `RepoAnalyzer`?

We're the primary user, so we can change/break it.

Reply:
219491c0 handles things in a more fine-grained manner